index.php

```diff
@@ -4,73 +4,121 @@
+    curl_close($ch);
+    if ($httpCode !== 401) {
+        error_log("ERROR: Authentication failed for {$cloudUsername}@{$cloudDomain}. Expected HTTP code 401, got $httpCode");
+        return False;
+    }
+
+    // extract the nonce from the response header
+    preg_match('/uthenticate: Digest ([0-9a-f]+)/', $response, $matches);
+
+    // if nonce is empty, return error
+    if (!isset($matches[1])) {
+        error_log("ERROR: Authentication failed for {$cloudUsername}@{$cloudDomain}. No nonce found in response");
+        return False;
+    }
+
+    // read nonce
+    $nonce = $matches[1];
+
+    // build the authentication token
+    $tohash = "$cloudUsername:$cloudPassword:$nonce";
+    $token = hash_hmac('sha1', $tohash, $cloudPassword);
+
+    return $token;
+}
+
 // login to cti using the cloud credentials and get the sip credentials using the /user/me API
-function getSipCredentials($cloudUsername, $cloudPassword, $cloudDomain, $isToken = false) {
+function getSipCredentials($cloudUsername, $cloudPassword, $cloudDomain, $isToken = false)
+{
     // Step 1: Authenticate and obtain the authentication token if isToken is false
     if (!$isToken) {
-        $authUrl = "https://$cloudDomain/webrest/authentication/login";
-        $authData = "username=".urlencode($cloudUsername)."&password=".urlencode($cloudPassword);
-    
-        $ch = curl_init($authUrl);
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-        curl_setopt($ch, CURLOPT_POST, true);
-        curl_setopt($ch, CURLOPT_POSTFIELDS, $authData);
-        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
-        curl_setopt($ch, CURLOPT_HEADER, true);
-        
-        $response = curl_exec($ch);
-        $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
-        
-        curl_close($ch);
-        if ($httpCode !== 401) {
-            error_log("ERROR: Authentication failed for {$cloudUsername}@{$cloudDomain}. Expected HTTP code 401, got $httpCode");
-            return False;
-        }
-
-        // Step 2: Extract the nonce from the response header
-        preg_match('/uthenticate: Digest ([0-9a-f]+)/', $response, $matches);
+        // get auth token
+        $token = getAuthToken($cloudUsername, $cloudPassword, $cloudDomain);
 
-        if (!isset($matches[1])) {
-            error_log("ERROR: Authentication failed for {$cloudUsername}@{$cloudDomain}. No nonce found in response");
-            return False;
-        }
-        
-        $nonce = $matches[1];
-        
-        // Step 3: Build the authentication token
-        $tohash = "$cloudUsername:$cloudPassword:$nonce";
-        $token = hash_hmac('sha1', $tohash, $cloudPassword);
-        if (getenv('DEBUG') && $_ENV['DEBUG'] === 'true') error_log("DEBUG: Token generated for {$cloudUsername}@{$cloudDomain}");
+        // print debug log
+        if (getenv('DEBUG') && $_ENV['DEBUG'] === 'true')
+            error_log("DEBUG: Token generated for {$cloudUsername}@{$cloudDomain}");
```

Comment:
Since we are doing also a refactor, let's transform this recurrent call to a function like _debug() or similar